### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'generated'

### ISSUE THREAD

**Issue by prostomarkeloff** [tortoise/tortoise-orm#246]
**Describe the bug**
Can't create a model with custom pk (id) data type (TextField).

Some part of pytest log:
```python
def __init__(self, **kwargs) -> None:
        # self._meta is a very common attribute lookup, lets cache it.
        meta = self._meta
>       self._saved_in_db = meta.pk_attr in kwargs and meta.pk.generated
E       AttributeError: 'NoneType' object has no attribute 'generated'

```

**To Reproduce**
```python
class UserModel(Model):
    id: str = fields.TextField(pk=True, generated=False)
    email: str = fields.CharField(index=True, unique=True, null=False, max_length=255)
    hashed_password: str = fields.CharField(null=False, max_length=255)
    is_active: bool = fields.BooleanField(default=True, null=False)
    is_superuser: bool = fields.BooleanField(default=False, null=False)

    class Meta:
        table = "user"
```

**Expected behavior**
All work.

**Comment by prostomarkeloff** [comment]
Oops. I have done my problem, but i still don't understand why this is happened.

**Comment by AEnterprise** [comment]
@prostomarkeloff you tried to pass ``generated`` to ``TextField`` but that field just doesn't support that property, ``TextField`` lets the database assign an autonumbered value to the field upon insertion

this is something that is only valid for numeric type fields like int, not for text

**Comment by prostomarkeloff** [comment]
> @prostomarkeloff you tried to pass `generated` to `TextField` but that field just doesn't support that property, `TextField` lets the database assign an autonumbered value to the field upon insertion
> 
> this is something that is only valid for numeric type fields like int, not for text

But i tried to pass the `False` value, which means that tortoise won't increment this.

**Comment by AEnterprise** [comment]
yes but it doesn't take a generated value at all because it isn't valid. it doesn't want ``generated``, true or false, because it can't do anything with it. in the context of of a ``TextField´´ the concept of auto-generating values doesn't exist, so it doesn't accept configuration options for it as there is no point in accepting options it can't do anything with

**Comment by grigi** [comment]
Actually, `TextField(pk=true)` should really be invalid, as we can't guarantee an index over a textfield.
e.g. It doesn't work with MySQL.

This is a valid bug as we should generate a better error message.

**Comment by grigi** [comment]
@prostomarkeloff Could you tell us how you solved the problem? Just for curiosity

**Comment by prostomarkeloff** [comment]
> @prostomarkeloff Could you tell us how you solved the problem? Just for curiosity

Interesting question. But i don't know ;) I just change some lines in my code and it has worked.

**Comment by prostomarkeloff** [comment]
> > @prostomarkeloff Could you tell us how you solved the problem? Just for curiosity
> 
> Interesting question. But i don't know ;) I just change some lines in my code and it has worked.

My code pass tests. You also can see more about it there: https://github.com/frankie567/fastapi-users/pull/59

**Comment by grigi** [comment]
Released v0.15.6 with DeprecationWarning when using TextField as primary key with suggestion to use CharField instead.
